File: Decorator/Java/CoffeeShowDecorator/src/CoffeeShopDecorator/CondimentDecorator.java
package CoffeeShopDecorator;

//Decorator
public abstract class CondimentDecorator extends Beverage 
{
    protected final Beverage wrappedBeverage;

    protected CondimentDecorator(Beverage beverage)
    {
        wrappedBeverage = beverage;
        description = "Undefined condiment";
    }

    @Override
    public double getPrice()
    {
        return wrappedBeverage.getPrice() + 0.5d;
    }

    @Override
    public String getDescription()
    {
        return String.format("%s, %s", wrappedBeverage.getDescription(), description);
    }
}

File: Decorator/C#/CoffeeShopDecorator/CondimentDecorator.cs
namespace CoffeeShopDecorator
{
    //Decorator
    public abstract class CondimentDecorator : Beverage 
    {
        protected readonly Beverage WrappedBeverage;

        protected CondimentDecorator(Beverage beverage)
        {
            WrappedBeverage = beverage;
            Description = "Undefined condiment";
        }

        public override decimal GetPrice()
        {
            return WrappedBeverage.GetPrice();
        }

        public override string GetDescription()
        {
            return WrappedBeverage.GetDescription();
        }
    }
}